frontend/server/src/Controllers/Problem.php

```diff
@@ -4764,52 +4766,27 @@ public static function getProblemListForSmarty(
             $minVisibility,
             $difficultyRange,
             $r->identity,
-            $r->user
-        );
-
-        $params = [
-            'query' => $keyword,
-            'language' => $language,
-            'order_by' => $orderBy,
-            'sort_order' => $sortOrder,
-            'tag' => $tags
-        ];
-
-        $pagerItems = \OmegaUp\Pager::paginateWithUrl(
-            $response['total'],
-            $pageSize,
-            $page ?: 1,
-            '/problem/list/',
-            5,
-            $params
+            $r->user,
+            /*$onlyQualitySeal=*/false,
+            /*$url=*/'/problem/list/'
         );
 
-        $tagData = [];
-        $allTags = self::getAllTagsFromCache();
-
-        foreach ($allTags as $tag) {
-            $tagData[] = ['name' => $tag->name];
-        }
-
         return [
             'smartyProperties' => [
                 'payload' => [
-                    'problems' => $response['results'],
+                    'problems' => $result['problems'],
                     'loggedIn' => !is_null($r->identity),
-                    'currentTags' => $tags,
-                    'pagerItems' => $pagerItems,
-                    'keyword' => $keyword,
-                    'mode' => $sortOrder,
-                    'column' => $orderBy,
-                    'language' => $language,
-                    'languages' => array_merge(
-                        ['all'],
-                        \OmegaUp\Controllers\Problem::VALID_LANGUAGES
-                    ),
-                    'modes' => \OmegaUp\Controllers\Problem::VALID_SORTING_MODES,
-                    'columns' => \OmegaUp\Controllers\Problem::VALID_SORTING_COLUMNS,
-                    'tags' => $tags,
-                    'tagData' => $tagData,
+                    'currentTags' => $result['currentTags'],
+                    'pagerItems' => $result['pagerItems'],
+                    'keyword' => $result['keyword'],
+                    'mode' => $result['mode'],
+                    'column' => $result['column'],
+                    'language' => $result['language'],
+                    'languages' => $result['languages'],
+                    'modes' => $result['modes'],
+                    'columns' => $result['columns'],
+                    'tags' => $result['tags'],
+                    'tagData' => $result['tags'],
```

Comment:
Me da la sensación de que aquí lo que esperas obtener es `tagData`. Eso me parece porque la anotación dice que espera un listado de Tags por nombre y ese parámetro es el que contiene eso.
```suggestion
                    'tagData' => $result['tagData'],
```
Mismo caso ocurre en las funciones:
- `getCollectionsDetailsByLevelForSmarty`
- `getCollectionsDetailsByAuthorForSmarty`